### PR TITLE
[ManagedIO] pass underlying transform URN as an annotation

### PR DESCRIPTION
**Pull request by ahmedabu98**

### CODE REVIEW

File: model/pipeline/src/main/proto/org/apache/beam/model/pipeline/v1/external_transforms.proto [hunk 1]
```diff
@@ -111,6 +111,15 @@ message BuilderMethod {
   bytes payload = 3;
 }
 
+message Annotations {
+  enum Enum {
+    CONFIG_ROW_KEY = 0 [(org.apache.beam.model.pipeline.v1.beam_constant) = "config_row"];
```
Comment (chamikaramj) [inline model/pipeline/src/main/proto/org/apache/beam/model/pipeline/v1/external_transforms.proto:116]: Let's add short descriptions regarding each of these.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/util/construction/PTransformTranslation.java [hunk 2]
```diff
@@ -522,11 +521,24 @@ public RunnerApi.PTransform translate(
...
+              ByteString.copyFromUtf8(identifier));
+          if (identifier.equals(MANAGED_TRANSFORM_URN)) {
+            Schema configSchema =
+                SchemaTranslation.schemaFromProto(payload.getConfigurationSchema());
+            Row configRow =
+                RowCoder.of(configSchema).decode(payload.getConfigurationRow().newInput());
+            String underlyingIdentifier =
+                MoreObjects.firstNonNull(
+                    configRow.getString("transform_identifier"), "unknown_identifier");
+            transformBuilder.putAnnotations(
```
Comment (chamikaramj) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/util/construction/PTransformTranslation.java:538]: Is there a valid case where "transform_identifier" would not be set ? If not we should just error out.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/util/construction/PTransformTranslation.java [hunk 3]
```diff
@@ -522,11 +521,24 @@ public RunnerApi.PTransform translate(
...
+              BeamUrns.getConstant(Annotations.Enum.SCHEMATRANSFORM_URN_KEY),
+              ByteString.copyFromUtf8(identifier));
+          if (identifier.equals(MANAGED_TRANSFORM_URN)) {
+            Schema configSchema =
+                SchemaTranslation.schemaFromProto(payload.getConfigurationSchema());
+            Row configRow =
+                RowCoder.of(configSchema).decode(payload.getConfigurationRow().newInput());
+            String underlyingIdentifier =
+                MoreObjects.firstNonNull(
+                    configRow.getString("transform_identifier"), "unknown_identifier");
```
Comment (chamikaramj) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/util/construction/PTransformTranslation.java:537]: Let's add unit tests to make sure that the annotations get added correctly.

**Review by chamikaramj — COMMENTED** [review]
Thanks!

**Comment by ahmedabu98** [comment]
Thanks for the view @chamikaramj, PTAL

**Review by chamikaramj — COMMENTED** [review]
Thanks. LGTM.

**Comment by ahmedabu98** [comment]
FYI @chamikaramj, I had to move the test out of `core` because it required a dependency on `managed`, which results in a circular dependency.

**Comment by ahmedabu98** [comment]
Test failures are unrelated

**Pull request merged**